examples/motoko_examples/simple-to-do/canisters/motoko/main.mo

```diff
@@ -0,0 +1,107 @@
+import Map "mo:base/HashMap";
+import Hash "mo:base/Hash";
+import Nat "mo:base/Nat";
+import Iter "mo:base/Iter";
+import Text "mo:base/Text";
+import Prim "mo:⛔";
+
+// Define the actor
+actor Assistant {
+
+  type ToDo = {
+    description: Text;
+    completed: Bool;
+  };
+  type PerfResult = {
+    wasm_body_only: Nat64;
+    wasm_including_prelude: Nat64;
+  };
+
+  func natHash(n : Nat) : Hash.Hash { 
+    Text.hash(Nat.toText(n))
+  };
+
+  var todos = Map.HashMap<Nat, ToDo>(0, Nat.equal, natHash);
+  var nextId : Nat = 0;
+  var perf_result: ?PerfResult = null;
+
+  public query func get_todos() : async [ToDo] {
+    let perf_start = Prim.performanceCounter(0);
+
+    let result = Iter.toArray(todos.vals());
+
+    let perf_end = Prim.performanceCounter(0);
+
+    perf_result := record_performance(perf_start, perf_end);
+
+    result
+  };
+
+  // Returns the ID that was given to the ToDo item
+  public func add_todo(description : Text) : async Nat {
+    let perf_start = Prim.performanceCounter(0);
+
+    let id = nextId;
```

Comment:
Why is this extra variable assignment here? The azle and motoko code should match more closely, I would say do whatever the motoko code originally did

Reply:
The original motoko code is
```
let id = nextId;
todos.put(id, { description = description; completed = false });
nextId += 1;
id
```
I believe the variable assignment is there because we need to return the value of the index as it was before the next_index gets incremented. I'll change the azle version to more closely match